### PR TITLE
SPSH-1247

### PR DESCRIPTION
**Pull request by YoussefBouch**
# Description

## Approval for review

- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

File: tests/Klasse.spec.ts [hunk 1]
```diff
@@ -133,4 +133,52 @@ test.describe(`Testfälle für die Administration von Klassen: Umgebung: ${proce
...
+    await test.step(`Klassenverwaltung öffnen und prüfen, dass jede Klasse eine Dienststellennummer hat`, async () => {
+        // Navigate to Klassenverwaltung
+        await startseite.card_item_schulportal_administration.click();
+        await menue.menueItem_AlleKlassenAnzeigen.click();
+
+        // Wait until the table is visible
+        await expect(klasseManagementView.text_h2_Klassenverwaltung).toHaveText("Klassenverwaltung");
+
+        // Define the selector for the 'Next' button
+        const nextPageButton = page.locator('.v-pagination__next button:not(.v-btn--disabled)');
```
Comment (jakobmadp) [inline tests/Klasse.spec.ts:151]: Please define locators in the page files. In this case FooterDataTable.page.ts. 
(playwright inspector helps to create locators: npx playwright codegen https://main.dev.spsh.dbildungsplattform.de/)
e.g. await page.getByLabel('Nächste Seite').click();

File: tests/Klasse.spec.ts [hunk 2]
```diff
@@ -133,4 +133,52 @@ test.describe(`Testfälle für die Administration von Klassen: Umgebung: ${proce
...
+
+        // Loop over each page
+        let hasNextPage = true;
+        while (hasNextPage) {
```
Comment (jakobmadp) [inline tests/Klasse.spec.ts:155]: I recommand not to use conditionals in ui testautomation (if possible). 
The loop doesn't work. The test will fail when the timeout is reached in 60s. The list of classes can have thousends of elements. 
Maybe its better to check the first 100 entries and the last 100 entries. Then you have a defined amount of classes to test.
Do something like that: 
1) Show first 100 entries
await page.getByTestId('klasse-table').getByRole('combobox').locator('i').click();
await page.getByRole('option', { name: '100' }).click();
2) check the entries
3) Go to the last page
await page.getByLabel('Letzte Seite').click();
4)  check the entries

File: tests/Klasse.spec.ts [hunk 3]
```diff
@@ -133,4 +133,52 @@ test.describe(`Testfälle für die Administration von Klassen: Umgebung: ${proce
...
+            // Get all rows in the current page's Klasse table
+            const rows = page.locator('table >> tbody >> tr');
+            const rowCount = await rows.count();
+
+            // Iterate over each row to check if Dienststellennummer and Klassenname are visible and not empty
+            for (let i = 0; i < rowCount; i++) {
```
Comment (jakobmadp) [inline tests/Klasse.spec.ts:161]: Maybe you can create a iteration and compare function in the page file KlasseManagementView.page.ts?
Something like that:
 public async iterateDataTable(xx) {
    xxx
  }

**Review by jakobmadp — CHANGES_REQUESTED** [review]

File: tests/Klasse.spec.ts (hunk 1 above)
Comment (YoussefBouch) [reply tests/Klasse.spec.ts:151]: Locators for last page are now defined in the main KlassenmanagementPage

File: tests/Klasse.spec.ts (hunk 2 above)
Comment (YoussefBouch) [reply tests/Klasse.spec.ts:155]: I check only for the first and last 100 entries in the table.

File: tests/Klasse.spec.ts (hunk 3 above)
Comment (YoussefBouch) [reply tests/Klasse.spec.ts]: Thought maybe it's not good practice to "expect" in a page. It's done.

**Review by jakobmadp — APPROVED** [review]

**Pull request merged**